gson/src/main/java/com/google/gson/Gson.java

```diff
@@ -573,13 +573,22 @@ public JsonElement toJsonTree(Object src, Type typeOfSrc) {
    * {@link #toJson(Object, Type)} instead. If you want to write out the object to a
    * {@link Writer}, use {@link #toJson(Object, Appendable)} instead.
    *
+   * For enums having  abstract methods , {@link Class#getClass()} will not return the Class
```

Comment:
This should be a comment inside the method. It is not a javadoc comment since Gson user can't do anything with this info.